tsdb/chunkenc/chunk.go

```diff
@@ -131,18 +131,18 @@ type Iterator interface {
 	// At returns the current timestamp/value pair if the value is a float.
 	// Before the iterator has advanced, the behaviour is unspecified.
 	At() (int64, float64)
-	// AtHistogram returns the current timestamp/value pair if the value is a
+	// AtHistogram returns a copy(!) of the current timestamp/value pair if the value is a
 	// histogram with integer counts. Before the iterator has advanced, the behaviour
 	// is unspecified.
-	// The method accepts an optional Histogram object which will be
+	// The method accepts an optional Histogram object which may be
```

Comment:
Doesn’t your PR enforce that the optional Histogram object **will** be reused?

Reply:
I'm not checking if the value is reused and also I see some examples of especially mock iterators that always return new objects.

Reply:
The documentation is then ambiguous. As a caller of `it.AtHistogram(h)`, I want to know whether the object corresponding to the pointer I passed has actually been reused or not, without having to go through all the implementations of the interface.
I think that we should use will as before, even because this way we kind of force new implementations to respect the contract: if they don't reuse the given object, and if something bad happens, then it is their fault.
Related to the mocked iterators, I don't know if they are so relevant.

Reply:
Sounds reasonable. I can also just update the mocks as well to be sure.

Reply:
Although there's one non mock that needs rework then: `func (c *concreteSeriesIterator) AtFloatHistogram(*histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {`